### PR TITLE
WooCommerce: Add support for fetching orders

### PR DESCRIPTION
**Pull request by aforcier**
Adds support for fetching orders from WooCommerce sites, along with some more base configuration for the WooCommerce FluxC plugin.

~Work in progress PR, not ready for review - also should be merged after #635 and #754 .~

* Adds handling for WooCommerce WP-API endpoints, with are added as test to `plugins/woocommerce/src/main/tools/wc-wp-api-endpoints.txt`, with the generated endpoint class being `WOOCOMMERCE`
* Adds the `ReleaseWCNetworkModule`, a counterpart to `ReleaseNetworkModule` but specific to WooCommerce network clients
* Builds out the stub `WCOrderModel` with the basic fields needed to populate the order list (more fields, such as line items and full customer info, will be added in a later PR)
* Adds the first set of actions for the WooCommerce plugin, in `WCOrderAction`. I ran into some trouble with the annotation processor when this was in Kotlin, so this ends up being the only Java part of the WooCommerce plugin 😢
* The `fetchOrders()` network call is using the Jetpack tunnel API call builder added in #635
* Some overhaul to the unit test configs was needed to support Robolectric in a Kotlin test. This work was extracted to #709. ~Until it's merged, there's a temporary commit in here (66c7080) adding the configs so that the tests work in this branch~ (rebased the commit out)
* `OrderStatus` contains constants for the default order statuses supported by WooCommerce. This will be expanded later to include any statuses added by major extensions we want to support. `OrderStatus` can't be an enum, because WooCommerce sites can have (by installing a plugin) arbitrary user-defined custom statuses. But, this still helps when filtering orders from FluxC by standard statuses.

### CODE REVIEW

**Comment by aforcier** [comment]
Since adding connected tests is getting somewhat involved, I'm add those separately (https://github.com/wordpress-mobile/WordPress-FluxC-Android/issues/751).

Also, I'd like to merge the `woocommerce-store` branch into `develop` soon as that's all been reviewed by a single person, and update this PR's base branch - so the `Not Ready For Merge` flag is on (thanks @AmandaRiu).

Ready for review @AmandaRiu!

**Comment by aforcier** [comment]
#754 has been merged, updated the base branch of this PR to `develop` 🎉

File: plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/persistence/OrderSqlUtils.kt [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    fun getOrdersForSite(site: SiteModel): List<WCOrderModel> {
+        return WellSql.select(WCOrderModel::class.java)
+                .where()
+                .equals(WCOrderModelTable.LOCAL_SITE_ID, site.id)
+                .endWhere()
+                .orderBy(WCOrderModelTable.DATE_CREATED, SelectQuery.ORDER_DESCENDING)
+                .asModel
+    }
+
+    fun getOrdersForSiteWithStatus(site: SiteModel, status: List<String>): List<WCOrderModel> {
```
[the remark this answers is not in the document]
Comment (aforcier) [reply plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/persistence/OrderSqlUtils.kt]: 👍 Done in 1a7d559.

> Do you think this is overcomplicating it?

Maybe a little, but that's okay. I think it's also possible we end up wanting to do different ordering rules when filtering by status vs without, but in that case we can just re-separate these two.

**Comment by AmandaRiu** [comment]
@aforcier Looks good! Merging.

**Pull request merged**